source/common/http/alternate_protocols_cache_impl.h

```diff
@@ -48,12 +50,22 @@ class AlternateProtocolsCacheImpl : public AlternateProtocolsCache {
   // Time source used to check expiration of entries.
   TimeSource& time_source_;
 
-  // Map from hostname to list of alternate protocols.
-  // TODO(RyanTheOptimist): Add a limit to the size of this map and evict based on usage.
-  std::map<Origin, std::vector<AlternateProtocol>> protocols_;
+  struct OriginHash {
+    size_t operator()(const Origin& origin) const {
+      size_t hash = std::hash<std::string>()(origin.scheme_) +
+                    37 * (std::hash<std::string>()(origin.hostname_) +
```

Comment:
37?

Reply:
Ah, I stole this from source/common/upstream/load_balancer_impl.h which seems to be based on a pattern that's referenced in Effective Java (a great book). Good enough?

Reply:
Also added a comment.